### PR TITLE
Enable full SAT for the Snowflake source

### LINKED ISSUE #19896: Enable SAT for hosted sources - Snowflake
Enable the SAT for the Snowflake source. See epic description for more context airbytehq/airbyte#19708.

### PR DESCRIPTION
**Pull request by DoNotPanicUA**
## What
Enable SAT for the Snowflake source

## Configured test instances
- sat_test_dataset

## Found issues
SAT related issues:
- The source returns time data with milliseconds, but the SAT treats this as a type violation. - :warning: **Shall we trim the data in the source?**
- The SAT fails when face non-typical values like `Nan`, `+inf` or `-inf`

### CODE REVIEW

File: airbyte-integrations/connectors/source-snowflake/acceptance-test-config.yml [hunk 1]
```diff
@@ -5,3 +5,19 @@ tests:
   spec:
     - spec_path: "src/test-integration/resources/expected_spec.json"
       config_path: "secrets/config.json"
+  connection:
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-snowflake/acceptance-test-config.yml:8]: Please use the latest acceptance-test-config format. [Example here.](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-zendesk-support/acceptance-test-config.yml#L1). I'd be curious to check the test output under `high` test strictness level, we might get this for free 😄

**Review by alafanechere — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-snowflake/acceptance-test-config.yml [hunk 2]
```diff
@@ -1,7 +1,29 @@
 # See [Source Acceptance Tests](https://docs.airbyte.com/connector-development/testing-connectors/source-acceptance-tests-reference)
 # for more information about how to configure these tests
 connector_image: airbyte/source-snowflake:dev
-tests:
+acceptance_tests:
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-snowflake/acceptance-test-config.yml:4]:
```suggestion
test_strictness_level: high
acceptance_tests:
```
I'd be curious to check the test output under high test strictness level, we might get this for free.

**Comment by DoNotPanicUA** [comment]
@alafanechere 
The CI run fails with the new format, but the local pass is fine.
Screenshot from 2022-12-07 17-05-27
any ideas on how to fix it?

**Comment by alafanechere** [comment]
> any ideas on how to fix it? 
I think you spotted a bug in my latest PR: https://github.com/airbytehq/airbyte/pull/20076
I'm going to take a look.

**Comment by DoNotPanicUA** [comment]
@alafanechere 
Looks like we can't set the strictness level as High because it requires the future cursor check.
This check doesn't work for the java sources because they use the old State massage handler, and it requires more deep investigation with the significant refactoring of the JDBC sources.

**Comment by alafanechere** [comment]
> Looks like we can't set the strictness level as High because it requires the future cursor check.
This check doesn't work for the java sources because they use the old State massage handler, and it requires more deep investigation with the significant refactoring of the JDBC sources.

It makes sense, thank you for trying!

**Review by alafanechere — CHANGES_REQUESTED** [review]
Requesting changes for the reasons explained here: https://github.com/airbytehq/airbyte/pull/19915#discussion_r1042325573

**Review by alafanechere — APPROVED** [review]

**Pull request merged**